agent.py

```diff
@@ -91,24 +94,36 @@ def call(self, states):
         performs the forward pass on a bunch of states, generate probabilities for taking each action
         uses 2 GRU to summarize sequential stock data, and a dense layer to classify to actions
 
-        :param states: a batch of input states [self.batch_sz * self.past_num * self.state_size]
-        :return a tensor of size [batch_sz * self.num_actions], representing the action probability of taking each
-                action
+        :param states: a batch of state tuples (self.batch_sz, ), where each tuple is (<price_history>, <portfolio>)
+                        where price_history is of shape (model.num_stocks, model.past_num, datum_size),
+                        and portfolio is of shape (model.num_stocks+1,)
+        :return a tensor of size (batch_sz, self.num_stock, 3), representing the action probability of taking each
+                action. The tensor always sums to 1 on the last dimension (the prob of buying/selling/holding a
+                particular stock on a particular day sums to one)
                 a tensor that summarize the states tensor of shape [batch_sz * self.actor_H2]
         """
-        gru_1_out_whole_seq, _ = self.actor_dropout_1(self.actor_gru_1(states))  # [batch_sz * past_num * actor_H1]
-        gru_2_out = self.actor_dropout_2(self.actor_gru_2(gru_1_out_whole_seq))  # [batch_sz * actor_H2]
-        actor_out = self.actor_dense(gru_2_out)  # [batch_sz * self.num_actions]
-        return actor_out, gru_2_out
+        # extract info from states
+        price_history = [state[0] for state in states]  # [batch_sz, num_stocks, past_num, datum_size]
+        price_history = tf.reshape(price_history, (-1, self.num_stocks * self.past_num, self.datum_size))  # TODO: this seems wrong
```

Comment:
i checked this on jupyter notebook (made a tensor and reshaped it this way), this seems to work.

Reply:
my main concern is whether the 2nd dim is justified to be of size `num_stocks * past_num`, because this would then become the `timestep` param of the GRU, which we may want to be just `past_num` (?)

sorry if my initial comments were not clear

Reply:
haha you are good

you know how we passed embedding_size as the timestep parameter for the language models assigment? I think num_stocks * past_num would accurately represent the "embedding" here because in this way the second dimension reflects a snapshot of our entire portfolio in the past number of days

Reply:
fixed - 66316bf